src/forwarder.ts

```diff
@@ -60,6 +60,7 @@ export async function addCloudWatchForwarderSubscriptions(
   aws: Aws,
   functionArn: CloudFormationObjectArn | string,
   integrationTesting: boolean | undefined,
+  subscribeToApiGatewayLogs: boolean,
```

Comment:
I think this function is getting a bit long/complicated. It takes five arguments and is almost 100 lines long.

A lot of that length/complexity was already there before this PR, but there's no doubt that this PR is adding more length/complexity. Therefore I think now would be a good time to do a refactor here.

First of all, five arguments is a lot -- a good guideline is to really try to avoid more than three arguments if possible. When I look at a call to this function: `await addCloudWatchForwarderSubscriptions(service as Service, aws, "my-func", false, true)`, it's pretty complicated, and it's not obvious what all of the arguments do, particularly the two booleans at the end.

One possible suggestion is switching the `integrationTesting` argument to an environment variable. That way we don't have to pass it around in the code.

I think there are also opportunities to move code out of this function and into smaller helper functions. I bet there are a lot of ways we could refactor this function to break it down into smaller pieces.

Another thought -- maybe we could have two separate functions -- one for the regular function subscriptions and one for the API Gateway-related ones?

I don't know if it's necessary to spend a huge amount of time totally re-doing everything here and implementing all of these suggestions, but a good rule of thumb would be leaving the code simpler than you found it. Otherwise things are just bound to get more and more complex over time.

Reply:
I just saw this but had already thought the same thing. I started to break down the function a bit and you can see that in the most recent commit. Let me know if you think it helps.